consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -432,12 +435,14 @@ impl VirtualStateProcessor {
                 let _ = self.notification_root().notify(Notification::VirtualDaaScoreChanged(
                     VirtualDaaScoreChangedNotification::new(new_virtual_state.daa_score),
                 ));
-                // TODO: fix the contents which is wrong in the following commented version
-                // let _ = self.notification_root().notify(Notification::VirtualChainChanged(VirtualChainChangedNotification::new(
-                //     new_virtual_state.ghostdag_data.mergeset_blues.clone(),
-                //     new_virtual_state.ghostdag_data.mergeset_reds.clone(),
-                //     Arc::new(new_virtual_state.accepted_tx_ids),
-                // )));
+                let chain_path = self.dag_traversal_manager.calculate_chain_path(prev_selected, new_selected);
+                let added_chain_blocks_acceptance_data =
+                    chain_path.added.iter().copied().map(|added| self.acceptance_data_store.get(added).unwrap()).collect_vec();
+                let _ = self.notification_root().notify(Notification::VirtualChainChanged(VirtualChainChangedNotification::new(
```

Comment:
Can you add a TODO to find way to read acceptance data from DB only if this subscription is actually active?